src/exo/pyparser.py

```diff
@@ -525,78 +557,102 @@ def parse_stmt_block(self, stmts):
+                    else:
+                        do_fresh_assignment = False
+
+                    # get the symbol corresponding to the name on the
+                    # left-hand-side
+                    if isinstance(s, (pyast.Assign, pyast.AugAssign)):
+                        if name_node.id not in self.locals:
+                            self.err(name_node, f"variable '{name_node.id}' undefined")
+                        nm = self.locals[name_node.id]
+                        if isinstance(nm, SizeStub):
+                            self.err(
+                                name_node,
+                                f"cannot write to " f"size variable '{name_node.id}'",
+                            )
+                        elif not isinstance(nm, Sym):
+                            self.err(
+                                name_node,
+                                f"expected '{name_node.id}' to "
+                                f"refer to a local variable",
+                            )
 
-                # generate the assignemnt or reduction statement
-                if do_fresh_assignment:
-                    rstmts.append(UAST.FreshAssign(nm, rhs, self.getsrcinfo(s)))
-                elif isinstance(s, pyast.Assign):
-                    rstmts.append(UAST.Assign(nm, idxs, rhs, self.getsrcinfo(s)))
-                elif isinstance(s, pyast.AugAssign):
-                    if not isinstance(s.op, pyast.Add):
-                        self.err(s, "only += reductions currently supported")
-                    rstmts.append(UAST.Reduce(nm, idxs, rhs, self.getsrcinfo(s)))
+                    # generate the assignemnt or reduction statement
+                    if do_fresh_assignment:
+                        rstmts.append(UAST.FreshAssign(nm, rhs, self.getsrcinfo(s)))
+                    elif isinstance(s, pyast.Assign):
+                        rstmts.append(UAST.Assign(nm, idxs, rhs, self.getsrcinfo(s)))
+                    elif isinstance(s, pyast.AugAssign):
+                        if not isinstance(s.op, pyast.Add):
+                            self.err(s, "only += reductions currently supported")
+                        rstmts.append(UAST.Reduce(nm, idxs, rhs, self.getsrcinfo(s)))
 
             # ----- For Loop parsing
             elif isinstance(s, pyast.For):
                 if len(s.orelse) > 0:
                     self.err(s, "else clause on for-loops unsupported")
 
                 self.push()
+
                 if not isinstance(s.target, pyast.Name):
                     self.err(s.target, "expected simple name for iterator variable")
-                itr = Sym(s.target.id)
-                self.locals[s.target.id] = itr
-                hi = self.parse_loop_cond(s.iter)
+
+                if self.is_fragment:
+                    itr = s.target.id
+                else:
+                    itr = Sym(s.target.id)
+                    self.locals[s.target.id] = itr
```

Comment:
Is this the same as `self.locals[s.target.id] = s.target.id.copy()`?

Reply:
Yup!

Reply:
Oh actually no, we need to create a symbol from string here

Reply:
Ok, cool. I wasn't sure... that class has too many similar-but-different ways to work with it.